middleware/database_session.py

```diff
@@ -12,13 +16,52 @@ def __init__(self, app):
         super().__init__(app)
 
     async def dispatch(self, request, call_next):
+        info = self._prepare_info(request)
 
-        session_token = general.get_ctx_token()
+        request.state.session_token = general.get_ctx_token()
```

Comment:
we dont need to set a session token for every request but rather for every task or thread.

In this instance this means that we should only remove the token in the _check_access function if an exception occured and we need to send and unauthorized response + remove line 26 (second token request)